src/coreclr/vm/jitinterface.cpp

```diff
@@ -9796,57 +9793,121 @@ CorInfoHFAElemType CEEInfo::getHFAType(CORINFO_CLASS_HANDLE hClass)
 
 /*********************************************************************/
 
-    // return the unmanaged calling convention for a PInvoke
-CorInfoUnmanagedCallConv CEEInfo::getUnmanagedCallConv(CORINFO_METHOD_HANDLE method)
+    // return the entry point calling convention for any of the following
+    // - a P/Invoke
+    // - a method marked with UnmanagedCallersOnly 
+    // - a function pointer with the CORINFO_CALLCONV_UNMANAGED calling convention.
+CorInfoCallConvExtension CEEInfo::getEntryPointCallConv(CORINFO_METHOD_HANDLE method, CORINFO_SIG_INFO* callSiteSig)
 {
     CONTRACTL {
         THROWS;
         GC_TRIGGERS;
         MODE_PREEMPTIVE;
     } CONTRACTL_END;
 
-    CorInfoUnmanagedCallConv result = CORINFO_UNMANAGED_CALLCONV_UNKNOWN;
+    CorInfoCallConvExtension callConv = CorInfoCallConvExtension::Managed;
 
     JIT_TO_EE_TRANSITION();
 
-    MethodDesc* pMD = NULL;
-    pMD = GetMethod(method);
-    _ASSERTE(pMD->IsNDirect());
+    _ASSERTE(method != nullptr || callSiteSig != nullptr);
 
-#ifdef TARGET_X86
-    EX_TRY
+    if (method)
     {
-        PInvokeStaticSigInfo sigInfo(pMD, PInvokeStaticSigInfo::NO_THROW_ON_ERROR);
+        MethodDesc* pMD = GetMethod(method);
+        if(pMD->IsNDirect())
+        {
+            EX_TRY
```

Comment:
This try/catch looks suspect. I know it was there even before your change, but it does not make it right. I would delete it and see what happens.